general-superstaq/general_superstaq/check/all_.py

```diff
@@ -16,13 +16,29 @@
     requirements,
 )
 
-
-def run(*args: str, sphinx_paths: Optional[List[str]] = None) -> int:
+CHECK_TYPE = Literal[
+    "configs",
+    "format",
+    "flake8",
+    "pylint",
+    "mypy",
+    "coverage",
+    "requirements",
+    "build_docs",
+]
```

Comment:
Not sure whether this is better to check at "compile time" (mypy) or runtime.  I could just instead define `CHECKS = ["configs", ...]` up here, accept a `skip: Optional[List[str]]` argument in `run` and then throw an error if `not all(entry not in CHECKS for entry in skip)`

Reply:
could we just add `parser.add_argument("--skip", choices=[...], ...)` to the arg parser?

Reply:
Ah nice yeah I'll do that!

Reply:
done